### DISCUSSION TITLE
Internal redirect in server middleware

### DISCUSSION THREAD

**Discussion by memic84** [nuxt/nuxt#20354] [Questions] [answered]
In Nuxt 2 it was possible to redirect from serverMiddleware. The docs say that you should not return and only add to the context in server middleware event.

When using the server middleware for a lot of redirects, **what the best way to redirect the user server-side** with a proper 301/302.

Route rules is not an option, because it's experimental.

**Comment by eggsy** [comment, marked as answer]
You can use `sendRedirect` method to redirect requests from `defineEventHandler` as stated in the [#sending-redirect](https://nuxt.com/docs/guide/directory-structure/server#sending-redirect) section of the documentation. I think the method is auto-imported, if not, it seems like it's a `h3` export, you may try to import from there in case you need it.

<img width="803" alt="image" src="https://user-images.githubusercontent.com/13917975/232833804-0e709b85-87b7-44a1-b66f-e15112e04128.png">

**Reply by memic84** [reply]
Thanks. I'll try this!

It worked for met this way:
```
event.node.res.writeHead(status_code, { Location: redirect_to });
event.node.res.end();
```

Probably not the best way to handle this in Nuxt.
